rest-api-spec/src/main/resources/rest-api-spec/test/index/90_flat_object.yml

```diff
@@ -0,0 +1,438 @@
+
+  # Range Query with dot path.
+  - do:
+      search:
+        body: {
+          _source: true,
+          query: {
+            "range": {
+              "catalog.catalogId": {
+                "gte": "c-0000",
+                "lte": "c-0006"
+              }
+            }
+          }
+        }
+
+  - length: { hits.hits: 1 }
+  - match: { hits.hits.0._source.catalog.catalogId: "c-0002" }
+
+  # Range Query without dot path.
+  - do:
+      search:
+        body: {
+          _source: true,
+          query: {
+            "range": {
+              "catalog": {
+                "gte": "c-0000",
+                "lte": "c-0006"
+              }
+            }
+          }
+        }
+
+  - length: { hits.hits: 1 }
+  - match: { hits.hits.0._source.catalog.catalogId: "c-0002" }
+
+  # Exists Query with dot path.
+  - do:
+      search:
+        body: {
+          _source: true,
+          query: {
+            "exists": {
+              "field": catalog.catalogId
+            }
+          }
+        }
+
+  - length: { hits.hits: 2 }
+
+  # Exists Query without dot path.
+  - do:
+      search:
+        body: {
+          _source: true,
+          query: {
+            "exists": {
+              "field": catalog
+            }
+          }
+        }
+
+  - length: { hits.hits: 2 }
+
+  # Query_string Query without dot path.
+  - do:
+      search:
+        body: {
+          _source: true,
+          query: {
+            "query_string": {
+              "fields": [ "catalog", "ISBN13" ],
+              "query": "John OR Mike"
+            }
+          }
+        }
+
+  - length: { hits.hits: 2 }
+  - match: { hits.hits.0._source.catalog.author.given: "Mike" }
+
+  # Query_string Query with dot path.
+  - do:
+      search:
+        body: {
+          _source: true,
+          query: {
+            "query_string": {
+              "fields": [ "catalog.author.given", "ISBN13" ],
+              "query": "John OR Mike"
+            }
+          }
+        }
+
+  - length: { hits.hits: 2 }
+  - match: { hits.hits.0._source.catalog.author.given: "Mike" }
+
+  # Simple_query_string Query without dot path.
+  - do:
+      search:
+        body: {
+          _source: true,
+          query: {
+            "simple_query_string" : {
+              "query": "Doe",
+              "fields": ["catalog", "ISBN13"]
+            }
+          }
+        }
+
+  - length: { hits.hits: 1 }
+  - match: { hits.hits.0._source.catalog.author.surname: "Doe" }
+
+
+  # Simple_query_string Query with dot path.
+  - do:
+      search:
+        body: {
+          _source: true,
+          query: {
+            "simple_query_string": {
+              "query": "Doe",
+              "fields": [ "catalog.author.surname", "ISBN13" ]
+            }
+          }
+        }
+
+  - length: { hits.hits: 1 }
+  - match: { hits.hits.0._source.catalog.author.surname: "Doe" }
+
+
+  # Wildcard Query with dot path.
+  - do:
+      catch: bad_request
```

Comment:
Please add matches for status and error cause: it should be very clear why request failed.

Reply:
Added match for status and reason in commit  
[Add tests for array, nested-arrary, number and float](https://github.com/opensearch-project/OpenSearch/pull/6507/commits/ec395b171a40344ac965705a6d44b1f75d6656a4)